src/herder/TxSetFrame.cpp

```diff
@@ -831,15 +830,30 @@ TxSetFrame::isGeneralizedTxSet() const
 }
 
 bool
-TxSetFrame::addTxsFromXdr(Hash const& networkID,
+TxSetFrame::addTxsFromXdr(Application& app,
                           xdr::xvector<TransactionEnvelope> const& txs,
                           bool useBaseFee, std::optional<int64_t> baseFee)
 {
     size_t oldSize = mTxs.size();
     mTxs.reserve(oldSize + txs.size());
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+    LedgerTxn ltx(app.getLedgerTxnRoot(), false,
+                  TransactionMode::READ_ONLY_WITHOUT_SQL_TXN);
+    auto ledgerVersion = ltx.loadHeader().current().ledgerVersion;
+    auto const& sorobanConfig =
+        app.getLedgerManager().getSorobanNetworkConfig(ltx);
+    ltx.rollback();
```

Comment:
What's the point of this `rollback`?

Reply:
Just wanted to get rid of ltx, but it's probably safe to let it live until the method is finished.